react-app/src/components/PortfolioScreen/PortfolioScreenAPI.ts

```diff
@@ -90,6 +81,71 @@ export const usePortfolioQuery = (): {
     [bankAPI, stakingAPI, distribution]
   );
 
+  const fetch = useCallback(
+    async (address?: string) => {
+      setRequestState(RequestStateLoading);
+
+      try {
+        if (address) {
+          if (!(await isValidAddress(address))) {
```

Comment:
i see some isValidAddress are being reimplemented, can we put it in somewhere common?